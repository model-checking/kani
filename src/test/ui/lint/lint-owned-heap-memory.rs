#![allow(dead_code)]
#![forbid(box_pointers)]


struct Foo {
    x: Box<isize> //~ ERROR type uses owned
}

fn main() {
    let _x: Foo = Foo { x : Box::new(10) };
    //~^ ERROR type uses owned
}
