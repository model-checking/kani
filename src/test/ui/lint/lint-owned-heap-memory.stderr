error: type uses owned (Box type) pointers: Box<isize>
  --> $DIR/lint-owned-heap-memory.rs:6:5
   |
LL |     x: Box<isize>
   |     ^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> $DIR/lint-owned-heap-memory.rs:2:11
   |
LL | #![forbid(box_pointers)]
   |           ^^^^^^^^^^^^

error: type uses owned (Box type) pointers: Box<isize>
  --> $DIR/lint-owned-heap-memory.rs:10:29
   |
LL |     let _x: Foo = Foo { x : Box::new(10) };
   |                             ^^^^^^^^^^^^

error: aborting due to 2 previous errors

